fedb/api-server.md

```diff
@@ -0,0 +1,157 @@
+- Start Date: 2021-04-23
+- Target Major Version: (2.2.0)
+- Reference Issues: 
+- Implementation PR: 
+
+# Add API Server
+
+# Summary
+
+HTTP interface is one of the most friendly interface and many developer like using http interface to develop their program. HTTP API also help us to show demo simply. So we want to support uing http api to access FEDB.
+
+# Detailed design
+
+
+## Interface design
+HTTP URL: http://ip:port/sql
+
+HTTP Method: POST
+
+### Put
+request
+```
+{
+    "method": "put",
+    "db": "db_name",
+    "table": "table_name",
+    "value": {
+        "field1": "value1",
+        "field2": 111,
+        "field3": 1.4   
+    }
+
+}
+```
+response
+```
+{
+    "code": 0,
+    "msg": "ok"
+}
+```
+
+### Execute procedure 
+reqeust
+```
+{
+    "method": "execute_procedure",
+    "db": "db_name",
+    "table": "table_name",
+    "value": {
+        "name": "procedure_name",
+        "input": {
+            "id":123,
+            "field1": "value1"
+        }
+    }
+
+}
+```
+response
+```
+{
+    "code": 0,
+    "msg": "ok",
+    "data": [{"field1" : "value1"}, {"field1": "value2"}]
```

Comment:
What's the expected size of output? Extracting schema might be an alternative.

Reply:
it seems a good idea when there are many records in output.